### PR TITLE
Add documentation for the new Ecovacs (Deebot) component and vacuum

### PR DESCRIPTION
**Pull request by OverloadUT**
**Description:**
New Ecovacs component which creates vacuums. Some extra disclaimers in this one due to how new the upstream library currently is

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#15520

## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/ecovacs.markdown [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+```yaml
+logger:
+  logs:
+    homeassistant.components.ecovacs: debug
+    homeassistant.components.vacuum.ecovacs: debug
+    sucks: debug
+```
+
+Warning: doing this will cause your authentication token to visibie in your log files. Be sure to remove any tokens and other authentication details from your log before posting them in an issue.
```
Comment (DubhAd) [inline source/_components/ecovacs.markdown]: Typo: *visibie* should be *visible*

File: source/_components/vacuum.ecovacs.markdown [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+{% raw %}
+```yaml
+# Example configuration.yaml entry
+sensor:
+  - platform: template
+    sensors:
+      vacuum_filter:
+        friendly_name: "Vacuum Filter Remaining Lifespan"
+        unit_of_measurement: '%'
+        value_template: "{{ states.vacuum.my_vacuum_id.attributes.component_filter }}"
```
Comment (DubhAd) [inline source/_components/vacuum.ecovacs.markdown]: `        value_template: "{{ state_attr('vacuum.my_vacuum_id', 'component_filter') }}"`
will avoid any errors if the component is still being set up

File: source/_components/vacuum.ecovacs.markdown [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+{% raw %}
+```yaml
+# Example configuration.yaml entry
+binary_sensor:
+  - platform: template
+    sensors:
+      vacuum_filter_replace:
+        friendly_name: "Vacuum Filter"
+        device_class: problem
+        value_template: "{{ states.vacuum.my_vacuum_id.attributes.component_main_brush <= 5 }}"
```
Comment (DubhAd) [inline source/_components/vacuum.ecovacs.markdown]: `        value_template: "{{ state_attr('vacuum.my_vacuum_id', 'component_main_brush') <= 5 }}"`

**Review by DubhAd — COMMENTED** [review]
One typo, two template recommendations

**Comment by OverloadUT** [comment]
@DubhAd Thanks; fixed!

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
Good job me merging while ha_release was not correct 🙈

**Pull request merged**